warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandler.java

```diff
@@ -1141,21 +1150,40 @@ protected void createShardFieldIndexColumn(RawRecordContainer event, Multimap<Bu
      *            flag for the delete count
      * @return a value
      */
-    private Value createUidArray(String uid, boolean isDeleted) {
-        
-        // Create a UID object for the Value
-        Builder uidBuilder = Uid.List.newBuilder();
-        uidBuilder.setIGNORE(false);
-        
-        if (isDeleted)
+    protected Value createUidArray(String uid, boolean isDeleted) {
+        return new Value(createUidList(uid, isDeleted).toByteArray());
+    }
+
+    /**
+     * Create a {@link Uid.List} given a uid and delete mode flag
+     *
+     * @param uid        an event uid
+     * @param deleteMode a flag
+     * @return a Uid.List
+     */
+    protected Uid.List createUidList(String uid, boolean deleteMode) {
+        Uid.List.Builder uidBuilder = Uid.List.newBuilder();
+
+        //  delete mode takes precedent over disabled uid creation
+        if (deleteMode) {
+            uidBuilder.setIGNORE(false);
             uidBuilder.setCOUNT(-1);
-        else
+            uidBuilder.addUID(uid);
+            return uidBuilder.build();
+        }
+
+        //  if uid creation is disabled
+        if (!getShardIndexCreateUids()) {
+            uidBuilder.setIGNORE(true);
             uidBuilder.setCOUNT(1);
-        
+            return uidBuilder.build();
+        }
+
+        //  else create a uid
+        uidBuilder.setIGNORE(false);
```

Comment:
This if/early return structure is basically mimicking an if/else if/else. I think the latter is a bit easier to follow:

```
if (isDelete)
  // set delete stuff
else if (isCreateUids)
  // set uid list stuff
else
  // set count only
return uidBuilder.build();
```

Reply:
I can certainly do that. Thoughts on order of precedent? I figured delete should take precedent over 'no uids', but I'm open to suggestions.